object_intern.go

```diff
@@ -25,7 +25,7 @@ type ObjectIntern struct {
 // nil if you want to use the default configuration, otherwise
 // pass in a custom configuration
 func NewObjectIntern(c *ObjectInternConfig) *ObjectIntern {
-	oi := &ObjectIntern{
+	oi := ObjectIntern{
```

Comment:
@robert-milan FYI this is a subtle go best practice. only get the address at the location where it should be clear that we intend to share the pointer to someone else.